stdlib/public/Concurrency/PartialAsyncTask.swift

```diff
@@ -370,6 +446,82 @@ extension ExecutorJob {
     unsafe _swiftJobRunOnTaskExecutor(UnownedJob(self), serialExecutor, taskExecutor)
   }
 }
+
+// Stack-disciplined job-local allocator support
+@available(SwiftStdlib 6.2, *)
+extension ExecutorJob {
+
+  /// Obtain a stack-disciplined job-local allocator.
+  ///
+  /// If the job does not support allocation, this property will be `nil`.
+  public var allocator: LocalAllocator? {
+    guard self.kind == .task else {
+      return nil
+    }
+
+    return LocalAllocator(context: self.context)
+  }
+
+  /// A job-local stack-disciplined allocator.
+  ///
+  /// This can be used to allocate additional data required by an
+  /// executor implementation; memory allocated in this manner will
+  /// be released automatically when the job is disposed of by the
+  /// runtime.
+  ///
+  /// N.B. Because this allocator is stack disciplined, explicitly
+  /// deallocating memory will also deallocate all memory allocated
+  /// after the block being deallocated.
```

Comment:
I think I had this question before already: that's now how we behave today, if you deallocate "in wrong order" we crash; are we really changing that? That was good for detecting we messed up somehow, while deallocating other things like this may hide bugs in stack discipline...?